src/Stratis.Bitcoin.Networks/BitcoinMain.cs

```diff
@@ -28,33 +28,51 @@ public BitcoinMain()
             this.MinRelayTxFee = 1000;
             this.CoinTicker = "BTC";
 
-            this.Consensus.SubsidyHalvingInterval = 210000;
-            this.Consensus.MajorityEnforceBlockUpgrade = 750;
-            this.Consensus.MajorityRejectBlockOutdated = 950;
-            this.Consensus.MajorityWindow = 1000;
-            this.Consensus.BuriedDeployments[BuriedDeployments.BIP34] = 227931;
-            this.Consensus.BuriedDeployments[BuriedDeployments.BIP65] = 388381;
-            this.Consensus.BuriedDeployments[BuriedDeployments.BIP66] = 363725;
-            this.Consensus.BIP34Hash = new uint256("0x000000000000024b89b42a942fe0d9fea3bb44ab7bd1b19115dd6a759c0808b8");
-            this.Consensus.PowLimit = new Target(new uint256("00000000ffffffffffffffffffffffffffffffffffffffffffffffffffffffff"));
-            this.Consensus.MinimumChainWork = new uint256("0x0000000000000000000000000000000000000000002cb971dd56d1c583c20f90");
-            this.Consensus.PowTargetTimespan = TimeSpan.FromSeconds(14 * 24 * 60 * 60); // two weeks
-            this.Consensus.PowTargetSpacing = TimeSpan.FromSeconds(10 * 60);
-            this.Consensus.PowAllowMinDifficultyBlocks = false;
-            this.Consensus.PowNoRetargeting = false;
-            this.Consensus.RuleChangeActivationThreshold = 1916; // 95% of 2016
-            this.Consensus.MinerConfirmationWindow = 2016; // nPowTargetTimespan / nPowTargetSpacing
-            this.Consensus.BIP9Deployments[BIP9Deployments.TestDummy] = new BIP9DeploymentsParameters(28, 1199145601, 1230767999);
-            this.Consensus.BIP9Deployments[BIP9Deployments.CSV] = new BIP9DeploymentsParameters(0, 1462060800, 1493596800);
-            this.Consensus.BIP9Deployments[BIP9Deployments.Segwit] = new BIP9DeploymentsParameters(1, 1479168000, 1510704000);
-            this.Consensus.CoinType = 0;
-            this.Consensus.DefaultAssumeValid = new uint256("0x000000000000000000174f783cc20c1415f90c4d17c9a5bcd06ba67207c9bc80"); // 518180
-            this.Consensus.CoinbaseMaturity = 100;
-            this.Consensus.PremineReward = Money.Zero;
-            this.Consensus.ProofOfWorkReward = Money.Coins(50);
-            this.Consensus.ProofOfStakeReward = Money.Zero;
-            this.Consensus.MaxReorgLength = 0;
-            this.Consensus.MaxMoney = 21000000 * Money.COIN;
+            var consensusFactory = new ConsensusFactory();
+
+            // Create the genesis block.
+            this.GenesisTime = 1231006505;
+            this.GenesisNonce = 2083236893;
+            this.GenesisBits = 0x1d00ffff;
+            this.GenesisVersion = 1;
+            this.GenesisReward = Money.Coins(50m);
+
+            Block genesisBlock = CreateBitcoinGenesisBlock(consensusFactory, this.GenesisTime, this.GenesisNonce, this.GenesisBits, this.GenesisVersion, this.GenesisReward);
+
+            this.Genesis = genesisBlock;
+
+            var consensus = new MutableConsensus();
```

Comment:
Can be changed in a way to use object initializer: 

```
var consensus = new MutableConsensus
{
    ConsensusFactory = consensusFactory,
    HashGenesisBlock = genesisBlock.GetHash(),
    SubsidyHalvingInterval = 210000,
    MajorityEnforceBlockUpgrade = 750,
    MajorityRejectBlockOutdated = 950,
    MajorityWindow = 1000,
    BuriedDeployments =
    {
        [BuriedDeployments.BIP34] = 227931,
        [BuriedDeployments.BIP65] = 388381,
        [BuriedDeployments.BIP66] = 363725
    },
    BIP34Hash = new uint256("0x000000000000024b89b42a942fe0d9fea3bb44ab7bd1b19115dd6a759c0808b8"),
    PowLimit = new Target(new uint256("00000000ffffffffffffffffffffffffffffffffffffffffffffffffffffffff")),
    MinimumChainWork = new uint256("0x0000000000000000000000000000000000000000002cb971dd56d1c583c20f90"),
    PowTargetTimespan = TimeSpan.FromSeconds(14 * 24 * 60 * 60),
    PowTargetSpacing = TimeSpan.FromSeconds(10 * 60),
    PowAllowMinDifficultyBlocks = false,
    PowNoRetargeting = false,
    RuleChangeActivationThreshold = 1916,
    MinerConfirmationWindow = 2016,
    BIP9Deployments =
    {
        [BIP9Deployments.TestDummy] = new BIP9DeploymentsParameters(28, 1199145601, 1230767999),
        [BIP9Deployments.CSV] = new BIP9DeploymentsParameters(0, 1462060800, 1493596800),
        [BIP9Deployments.Segwit] = new BIP9DeploymentsParameters(1, 1479168000, 1510704000)
    },
    CoinType = 0,
    DefaultAssumeValid = new uint256("0x000000000000000000174f783cc20c1415f90c4d17c9a5bcd06ba67207c9bc80"),
    CoinbaseMaturity = 100,
    PremineReward = Money.Zero,
    ProofOfWorkReward = Money.Coins(50),
    ProofOfStakeReward = Money.Zero,
    MaxReorgLength = 0,
    MaxMoney = 21000000 * Money.COIN
};
```

But I guess it's personal preference so don't change if you consider it to be better as it is.

Reply:
Same stands for other networks.